File: resources/information/technologies/java/lang/classes/basic/classMembers/src/Account.java
public class Account
{
   public Account(String accountNumber)
   {
     this( accountNumber, 0.0 );
   }
 
   public Account(String accountNumber, double initialBalance)
   {
     ++numInstances;
     accNo = accountNumber;
     balance   = initialBalance;
   }
 
   public void credit(double amount)
   {
     balance += amount;
   }
 
   public void debit(double amount)
   {
     credit(-amount);
   }
 
   public double getBalance()
   {
     return balance;
   }
 
   public String getAccountNumber()
   {
    return accountNo;
   }
 
   public String toString()
   {
     return accountNo + ": " + balance;
   }
 
   public void finalize()
   {
     System.out.println(accountNo + " is finalized.");
     --numInstances;
   }

   public static int getNumberOfInstances() 
   {
     return numInstances;
   }
 
   private String accountNo;
   private double balance;

   private static int numInstances = 0;
}

File: projects/solms/training/courses/c++/book/NameSpaces/Programs/Account.cpp
\begin{verbatim}
#include "Account.h"
#include <iostream>

using namespace std;

SolmsTraining::finance::Account::Account()
  : _balance(50) {}

SolmsTraining::finance::Account::Account(double initialBalance)
  : _balance(initialBalance) {}

SolmsTraining::finance::Account::~Account()
{
  cout << "I, " << this << ", am destroyed." << endl;
}

void SolmsTraining::finance::Account::credit(double amount)
{
  _balance += amount;
}

void SolmsTraining::finance::Account::debit(double amount)
{
  _balance -= amount;
}

double SolmsTraining::finance::Account::balance() {return _balance;}
\end{verbatim}
